sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/StateHolder.java

```diff
@@ -134,13 +134,25 @@ public static void setNextForcedRefresh(Duration refreshPeriod) {
      * @param properties Provider properties for min and max backoff periods.
      */
     static void resetAll(Duration refreshInterval, AppConfigurationProviderProperties properties) {
-        nextForcedRefresh = CalculatedBackoffTime.calculateBefore("client", nextForcedRefresh, refreshInterval, properties);
+        if (refreshInterval != null) {
+            Instant newForcedRefresh = CalculatedBackoffTime.calculateBefore(nextForcedRefresh,
+                clientRefreshAttempts, refreshInterval, properties);
+
+            if (newForcedRefresh.compareTo(nextForcedRefresh) != 0) {
+                clientRefreshAttempts += 1;
```

Comment:
After a few failed refresh attempts, we might get the same backoff time each time. This is because our max backoff is either 10 mins or user specified refresh interval, both of which are constant for the life of an application. So after a few failed refreshes, its possible that backoff time is always 10 minutes (or always user specified refresh interval). In that case, the clientRefreshAttempts will not be incremented.